src/main/java/com/synopsys/integration/alert/web/security/authentication/ldap/LdapManager.java

```diff
@@ -108,39 +128,49 @@ private DirContextAuthenticationStrategy createAuthenticationStrategy(final Ldap
         return strategy;
     }
 
-    private void updateAuthenticationProvider() throws AlertLDAPConfigurationException {
-        final LdapAuthenticator authenticator = createAuthenticator();
-        final LdapAuthoritiesPopulator authoritiesPopulator = createAuthoritiesPopulator();
+    private void updateAuthenticationProvider(final ConfigurationModel configurationModel) throws AlertLDAPConfigurationException {
+        final LdapAuthenticator authenticator = createAuthenticator(configurationModel);
+        final LdapAuthoritiesPopulator authoritiesPopulator = createAuthoritiesPopulator(configurationModel);
         authenticationProvider = new LdapAuthenticationProvider(authenticator, authoritiesPopulator);
     }
 
-    private LdapAuthenticator createAuthenticator() throws AlertLDAPConfigurationException {
+    private LdapAuthenticator createAuthenticator(final ConfigurationModel configurationModel) throws AlertLDAPConfigurationException {
         final BindAuthenticator authenticator = new BindAuthenticator(contextSource);
         try {
-
-            authenticator.setUserSearch(createLdapUserSearch(contextSource));
-            authenticator.setUserDnPatterns(currentConfiguration.getUserDNPatternArray());
-            authenticator.setUserAttributes(currentConfiguration.getUserAttributeArray());
+            final String[] userDnArray = createSetFromCSV(getFieldValueOrEmpty(configurationModel, SettingsDescriptor.KEY_LDAP_USER_DN_PATTERNS));
+            final String[] userAttributeArray = createSetFromCSV(getFieldValueOrEmpty(configurationModel, SettingsDescriptor.KEY_LDAP_USER_ATTRIBUTES));
+            authenticator.setUserSearch(createLdapUserSearch(configurationModel, contextSource));
+            authenticator.setUserDnPatterns(userDnArray);
+            authenticator.setUserAttributes(userAttributeArray);
             authenticator.afterPropertiesSet();
         } catch (final Exception ex) {
             throw new AlertLDAPConfigurationException("Error creating LDAP authenticator", ex);
         }
         return authenticator;
     }
 
-    private LdapAuthoritiesPopulator createAuthoritiesPopulator() {
-        final DefaultLdapAuthoritiesPopulator authoritiesPopulator = new DefaultLdapAuthoritiesPopulator(contextSource, currentConfiguration.getGroupSearchBase());
-        authoritiesPopulator.setGroupSearchFilter(currentConfiguration.getGroupSearchFilter());
-        authoritiesPopulator.setGroupRoleAttribute(currentConfiguration.getGroupRoleAttribute());
-        authoritiesPopulator.setRolePrefix(currentConfiguration.getRolePrefix());
+    private String[] createSetFromCSV(final String commaSeparatedString) {
+        return StringUtils.split(commaSeparatedString, ",");
```

Comment:
This is a little confusing since the method name implies it is creating a set not an array